src/sentry/search/events/builder.py

```diff
@@ -0,0 +1,272 @@
+        if "" in values:
+            values.remove("")
+            operator = Op.IS_NULL if search_filter.operator == "=" else Op.IS_NOT_NULL
+            env_conditions.append(Condition(environment, operator))
+        if len(values) == 1:
+            operator = Op.EQ if search_filter.operator in EQUALITY_OPERATORS else Op.NEQ
+            env_conditions.append(Condition(environment, operator, values.pop()))
+        elif values:
+            operator = Op.IN if search_filter.operator in EQUALITY_OPERATORS else Op.NOT_IN
+            env_conditions.append(Condition(environment, operator, values))
+        if len(env_conditions) > 1:
+            return Or(conditions=env_conditions)
+        else:
+            return env_conditions[0]
+
+    def resolve_select(self, selected_columns: List[str]) -> None:
+        for field in selected_columns:
+            if isinstance(field, str) and field.strip() == "":
+                continue
+            field = self.resolve_field(field)
+            if isinstance(field, Column) and field not in self.columns:
+                self.columns.append(field)
+
+    def resolve_field(self, field) -> Union[Column, Function]:
+        if not isinstance(field, str):
+            raise InvalidSearchQuery("Field names must be strings")
+
+        match = is_function(field)
+        if match:
+            raise NotImplementedError(f"{field} not implemented in snql field parsing yet")
+
+        if field in FIELD_ALIASES:
+            raise NotImplementedError(f"{field} not implemented in snql field parsing yet")
+
+        tag_match = TAG_KEY_RE.search(field)
+        field = tag_match.group("tag") if tag_match else field
+
+        if VALID_FIELD_PATTERN.match(field):
+            if field in self.field_allowlist:
+                return self.column(field)
+            else:
+                raise NotImplementedError(f"{field} not implemented in snql field parsing yet")
+        else:
+            raise InvalidSearchQuery(f"Invalid characters in field {field}")
+
+    @property
+    def select(self) -> Optional[List[SelectType]]:
+        return [*self.aggregates, *self.columns]
+
+    @property
+    def groupby(self) -> Optional[List[SelectType]]:
+        if self.aggregates:
+            return self.columns
+        else:
+            return []
+
+    @property
+    def orderby(self) -> List[OrderBy]:
+        validated = []
+        for column in self.orderby_columns:
+            bare_column = self.column(column.lstrip("-"))
+            direction = Direction.DESC if column.startswith("-") else Direction.ASC
+
+            if bare_column in self.columns:
+                validated.append(OrderBy(bare_column, direction))
+                continue
+
+            # TODO: orderby aggregations
+
+            # TODO: orderby field aliases
+
+        if len(validated) == len(self.orderby_columns):
+            return validated
+
+        raise InvalidSearchQuery("Cannot order by a field that is not selected.")
```

Comment:
You can actually. I made this error when I was writing the SnQL parser.

Reply:
Gonna leave this as-is for now to keep functionality consistent, but I'll add a comment explaining that this error is no longer valid.